Sources/SWBBuildSystem/BuildOperation.swift

```diff
@@ -1601,6 +1602,33 @@ internal final class OperationSystemAdaptor: SWBLLBuild.BuildSystemDelegate, Act
         }
     }
 
+    func cleanupGlobalModuleCache() {
+        let settings = operation.requestContext.getCachedSettings(operation.request.parameters)
+        if settings.globalScope.evaluate(BuiltinMacros.KEEP_GLOBAL_MODULE_CACHE_DIRECTORY) {
+            return // Keep the cache directory.
+        }
+
+        let cachePath = settings.globalScope.evaluate(BuiltinMacros.MODULE_CACHE_DIR)
+        guard !cachePath.isEmpty, operation.fs.exists(cachePath) else {
+            return
+        }
+
+        let signatureCtx = InsecureHashContext()
+        signatureCtx.add(string: "CleanupGlobalModuleCache")
+        signatureCtx.add(string: cachePath.str)
+        let signature = signatureCtx.signature
+
+        withActivity(ruleInfo: "CleanupGlobalModuleCache \(cachePath.str)", executionDescription: "Cleanup global module cache at \(cachePath)", signature: signature, target: nil, parentActivity: nil) { activity in
+            do {
+                try operation.fs.removeDirectory(cachePath)
+            } catch {
+                // Log error but do not fail the build.
+                emit(diagnostic: Diagnostic.init(behavior: .warning, location: .unknown, data: DiagnosticData("Error cleaning up \(cachePath): \(error.localizedDescription)")), for: activity, signature: signature)
```

Comment:
Phrase as "failed to remove" rather than "error cleaning up"?

Using the word "error" in a warning message could be confusing.

Reply:
sure, makes sense